### DISCUSSION TITLE
Two versions of tidy

### DISCUSSION THREAD

**Discussion by bikerpaul68** [Homebrew/discussions#4964] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: ab6a7ed5a122b6d4f9d441ef76c86032398ced5f
Last commit: 4 days ago
Core tap JSON: 09 Dec 14:21 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.43.0 => /usr/local/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 12.7.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  libav

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ilmbase
  libav
  openssl@1.1
  youtube-dl

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  xpdf
  unbound
  gnupg
```

### Description of issue

I have installed tidy-html5. It is installed in usr/local/Cellar/tidy-html5/5.8.0/bin/ with a symlink to /usr/local/bin / and /usr/local/bin is before /usr/bin in $PATH. Running ./tidy -v from its location gives the correct version but tidy -v from there or anywhere else gives the old Mac version of 2006. The only solution I have found on the internet is to use brew untap but this is no longer possible. I would like to use the new version with Bluefish to verify my HTML code. Can anyone help? TIA.

**Comment by ZhongRuoyu** [comment]
What is the output of `which -a tidy`?

**Comment by bikerpaul68** [comment]
/usr/local/bin/tidy
/usr/bin/tidy

I posted that I had solved the problem by running hash -r but my answer doesn't seem to have appeared. Anyway thank you for your reply.

**Comment by bikerpaul68** [comment]
OK, solved it myself after another web search. The problem isn't with Homebrew. I hadn't realized that bash remembers locations of utilities in a shell session. Running hash -r solved the problem.
